rmf_task/include/rmf_task/Task.hpp

```diff
@@ -78,10 +78,39 @@ class Task::Booking
   ///
   /// \param[in] automatic_
   ///   Whether this booking was automatically generated
+  ///
+  /// TODO(AC): Deprecate this constructor
+  Booking(
+    std::string id_,
+    rmf_traffic::Time earliest_start_time_,
+    ConstPriorityPtr priority_,
+    bool automatic_ = false);
+
+  /// Constructor
+  ///
+  /// \param[in] id_
+  ///   The identity of the booking
+  ///
+  /// \param[in] earliest_start_time_
+  ///   The earliest time that the task may begin
+  ///
+  /// \param[in] request_time_
+  ///   The time that this task was booked
+  ///
+  /// \param[in] priority_
+  ///   The priority of the booking
+  ///
+  /// \param[in] requester_
+  ///   The identifier of the entity that requested this task
+  ///
+  /// \param[in] automatic_
+  ///   Whether this booking was automatically generated
   Booking(
     std::string id_,
     rmf_traffic::Time earliest_start_time_,
+    rmf_traffic::Time request_time_,
     ConstPriorityPtr priority_,
+    std::string requester_,
```

Comment:
Nit: We should drop the trailing underscores from all the args

Reply:
hahaha gotcha. I was wondering about that too but decided to follow

Reply:
I'm not very certain about this, but do you know if dropping the underscore might cause ABI breaks due to any symbol changes?

Reply:
nah, the function signature is made up of the function name and the types of parameters, the names of the parameters can be changed without issues